plugin/cog/cog-commons/src/main/java/it/geosolutions/imageioimpl/plugins/cog/CogTileInfo.java

```diff
@@ -0,0 +1,147 @@
+/*
+ *    ImageI/O-Ext - OpenSource Java Image translation Library
+ *    http://www.geo-solutions.it/
+ *    http://java.net/projects/imageio-ext/
+ *    (C) 2019, GeoSolutions
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License as published by the Free Software Foundation;
+ *    either version 3 of the License, or (at your option) any later version.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ */
+package it.geosolutions.imageioimpl.plugins.cog;
+
+import java.util.Map;
+import java.util.TreeMap;
+
+/**
+ * This is a utility class that stores all of the required tile range metadata for a given read operation.
+ *
+ * @author joshfix
+ * Created on 2019-08-28
+ */
+public class CogTileInfo {
+
+    protected int headerLength = DEFAULT_HEADER_LENGTH;
+    protected Map<Integer, TileRange> tileRanges = new TreeMap<>();
+    public static final int HEADER_TILE_INDEX = -100;
+    public static final int DEFAULT_HEADER_LENGTH = 16384;
```

Comment:
I personally have not seen this and decided on 16K because it's the default for GDAL COG reads as well.  I would assume if a TIFF had enough overviews and enough tiles, you could eventually break past 16K, but it seems unlikely to me?  I just realized that I made an effort to make the initial headerLength configurable, but provided no real mechanism to allow an external user to set it.  I'm thinking I could just add it as a field to `CogImageReadParam`.... thoughts?